pkgs/build-support/kernel/modules-closure.sh

```diff
@@ -68,7 +68,11 @@ done
 
 mkdir -p $out/lib/firmware
 for module in $(cat closure); do
-    for i in $(modinfo -F firmware $module); do
+    for i in $(modinfo -b $kernel --set-version "$version" -F firmware $module); do
+        # $ modinfo -F firmware unix
+        # name:           unix
+        if test "$i" = "name:"; then break; fi
```

Comment:
I somewhat agree `modinfo` acts weird, piping it to `| grep -v` actually sounds better. Let me know.

Reply:
Yeah, while the kmod fix is my main preference, my preferred _workaround_ is the grep approach, since otherwise we'll fail to load firmware for any built-in modules that require it.